Sources/SwiftDocC/Infrastructure/Symbol Graph/ExtendedTypesFormatTransformation.swift

```diff
@@ -0,0 +1,530 @@
+        
+        let contextOfRelationships = synthesizeSecondaryExtendedTypeSymbols(&extendedTypeSymbols)
+        
+        redirect(\.target, of: &memberOfRelationships, using: extensionBlockToExtendedTypeMapping)
+        
+        redirect(\.source, of: &conformsToRelationships, using: extensionBlockToExtendedTypeMapping)
+        
+        attachDocComments(to: &extendedTypeSymbols.values, using: { (target) -> [SymbolGraph.Symbol] in
+            guard let relevantExtensionBlockSymbols = extendedTypeToExtensionBlockMapping[target.identifier.precise]?.compactMap({ id in extensionBlockSymbols[id] }).filter({ symbol in symbol.docComment != nil }) else {
+                return []
+            }
+            
+            // we sort the symbols here because their order is not guaranteed to stay the same
+            // across compilation processes and we always want to choose the same doc comment
+            // in case there are multiple candidates with maximum number of lines
+            if let winner = relevantExtensionBlockSymbols.sorted(by: \.identifier.precise).max(by: { a, b in (a.docComment?.lines.count ?? 0) < (b.docComment?.lines.count ?? 0) }) {
+                return [winner]
+            } else {
+                return []
+            }
+        })
+        
+        symbolGraph.relationships.append(contentsOf: memberOfRelationships)
+        symbolGraph.relationships.append(contentsOf: conformsToRelationships)
+        symbolGraph.relationships.append(contentsOf: contextOfRelationships)
+        extendedTypeSymbols.values.forEach { symbol in symbolGraph.symbols[symbol.identifier.precise] = symbol }
+        
+        try synthesizeExtendedModuleSymbolAndDeclaredInRelationships(on: &symbolGraph,
+                                                                      using: extendedTypeSymbols.values.filter { symbol in symbol.pathComponents.count == 2 }.map(\.identifier.precise),
+                                                                      moduleName: moduleName)
+        
+        return true
+    }
+
+    /// Tries to obtain `docComment`s for all `targets` and copies the documentation from sources to the target.
+    ///
+    /// Iterates over all `targets` calling the `source` method to obtain a list of symbols that should serve as sources for the target's `docComment`.
+    /// If there is more than one symbol containing a `docComment` in the compound list of target and the list returned by `source`, `onConflict` is
+    /// called iteratively on the (modified) target and the next source element.
+    private static func attachDocComments<T: MutableCollection>(to targets: inout T,
+                                                                using source: (T.Element) -> [SymbolGraph.Symbol],
+                                                                onConflict resolveConflict: (_ old: T.Element, _ new: SymbolGraph.Symbol)
```

Comment:
This function has a bit of unused functionality and removing it can simplify both the function and its call site. 

This is private and only called from one place which doesn't pass a value for the `onConflict` argument and which either returns a source with 0 or 1 elements. Removing the `onConflict` argument and changing the `source` argument type to 
```diff
+ using sourceDocComment: (T.Element) -> SymbolGraph.LineList?
- using source:           (T.Element) -> [SymbolGraph.Symbol]
```
means that the for loop can be replaced with just the assignment 
``` swift
target.docComment = sourceDocComment(target)
```
(since the `guard` statement ensures that the target doc comment is `nil`.) 

With the change to return the doc comment directly from the closure, the call site `compactMap` can chain to get the doc comment which removes the optional types in the check for the longest doc comment.  

Since the extension declarations are sorted based on their precise identifiers which is also the keys of `extensionBlockSymbols` I think that the call site can be written as:
```swift
attachDocComments(to: &extendedTypeSymbols.values) { target in
    return extendedTypeToExtensionBlockMapping[target.identifier.precise]?
        .sorted() // The order of symbols in not stable across different builds.
        .compactMap({ extensionBlockSymbols[$0]?.docComment })
        .max(by: { lhs, rhs in lhs.lines.count < rhs.lines.count })
}
```

Reply:
This function is more complex than needed because I plan on iterating on the aggregation logic once the basics are complete. I'll create an issue for that and link it here when I find time.